FreeRTOS-Plus/Source/FreeRTOS-Plus-TCP/FreeRTOS_Sockets.c

```diff
@@ -1397,6 +1397,12 @@ FreeRTOS_Socket_t *pxSocket;
 	( void ) uxOptionLength;
 
 	configASSERT( xSocket != NULL );
+	
```

Comment:
The title of this PR is "Add Full-TCP test suite", and the description does not include the fact that it modifies FreeRTOS+TCP behavior.  Changes to FreeRTOS_Sockets.c (and in fact any FreeRTOS+TCP core files) should be in a separate PR, since these affect more than just testing.